perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuditMessagesManagerImpl.java

```diff
@@ -203,6 +203,16 @@ public List<AuditMessage> getMessagesByCount(PerunSession perunSession, int coun
 		}
 	}
 
+	@Override
+	public List<AuditMessage> getMessagesByIdAndCount(PerunSession perunSession, int id, int count) {
+		int lowestId = id - count + 1;
+		try {
+			return jdbc.query("select " + auditMessageMappingSelectQuery + " from auditer_log where id between ? and ? order by id desc", AUDIT_MESSAGE_MAPPER, lowestId, id);
```

Comment:
just a thought -  i see there's similar problem with `getMessagesByCount` method - by the name of the method, wouldn't it be better to try to return the exact count of the messages than subtract the ids? or rather change the name to `getMessagesByIdsRange`?
```suggestion
			return jdbc.query("select " + auditMessageMappingSelectQuery + " from auditer_log where id<=? order by id desc limit ?", AUDIT_MESSAGE_MAPPER, id, count);
```

Reply:
Good point. Changed so it is using `limit` instead of `between`.